### DISCUSSION TITLE
reset isSuccess

### DISCUSSION THREAD

**Discussion by daggala** [TanStack/query#1014] [General]
I have a form where I am using isSuccess in my useMutation hook. When I finish saving the form I display a success message. Then when I would like to edit the form again, then I would like to reset isSuccess to false like in the beginning, display no success message and save the form again. Is there a way to do this? I saw the reset function but it only resets data and error.

**Comment by denisborovikov** [comment]
The `isSuccess` from `useMutation` hook shows the result of the mutation, not the state of your form. It's a different state. Yes, it derives from the hook's state at the beginning, but lately, you use it separately. Use a different state variable.

Something like that
```js
const [isFormSuccess, setIsFormSuccess] = React.useState(false);
const {isSuccess} = useMutation(...);

// Sync values of `isFormSuccess` and `isSuccess` if `isSuccess` changes
React.useEffect(
  () => {setIsFormSuccess(isSuccess)},
  [isSuccess]
)


function resetForm() {
  setIsFormSuccess(false);
}

....

<button onClick={resetForm}>Reset</button>
```

**Comment by felipemaccari** [comment]
you can use the 'reset' that the useMutation provides!

just call it whenever you want

`const [mutate, {isSuccess, isLoading, reset}) = useMutation(yourMutation)`

**Reply by sergnio** [reply]
This is awesome! I had no idea about `reset` and it's perfect for my usecase.
Thanks for this answer
